community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```diff
@@ -460,26 +459,62 @@ else if ( state == State.RUNNING || state == State.STARTING )
         }
     }
 
-    private void apply( IndexUpdates updates, IndexUpdateMode updateMode )
+    private void apply( Iterable<IndexEntryUpdate<LabelSchemaDescriptor>> updates, IndexUpdateMode updateMode )
```

Comment:
Yeah, I acted on an old TODO here, but it didn't turn out very well. Reverted to old behavior using `IndexUpdaterMap`, which achieves roughly what you propose here.